### PR TITLE
Add SELinux PSP

### PR DESCRIPTION
**Pull request by EmandM**

### CODE REVIEW

File: Policies/Kubernetes/selinux/azurepolicy.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      },
+      "excludedNamespaces": {
+        "type": "Array",
+        "metadata": {
+          "displayName": "Namespace exclusions",
+          "description": "List of Kubernetes namespaces to exclude from policy evaluation. Providing a value for this parameter is optional."
+        },
+        "defaultValue": []
+      },
+      "allowedSELinuxLevel": {
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/selinux/azurepolicy.json:33]: Looking at REGO tests, I understand not all SELinux parameters have to be provided. But I see the SELinux parameters defined in Azure policy do not have default value which means all the parameter values are required during assignment. Can you please double check the below rego test scenario through Azure Policy?

test_input_seLinux_options_allowed_in_list_subset {
    input := { "review": input_review, "parameters": input_parameters_in_list_subset}
    results := violation with input as input
    count(results) == 0
}

Specifying only subset of properties is a valid scenario and we need to support it through Azure Policy. For example, look at a sample PSP shared by a customer in https://github.com/Azure/azure-arc-kubernetes-preview/issues/100. You can see only 'rule' property defined in SELinux.

File: Policies/Kubernetes/selinux/rego/src_test.rego [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+test_input_seLinux_option_not_allowed_not_in_list {
+    input := { "review": input_review, "parameters": input_parameters_not_in_list}
+    results := violation with input as input
+    count(results) == 1
+}
+
+test_input_seLinux_options_empty {
+    input := { "review": input_review, "parameters": input_parameters_empty}
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/selinux/rego/src_test.rego:22]: As input_parameters_empty is a valid case and it means SELinux properties should not be defined, we need to specially mention this in parameter description.

File: Policies/Kubernetes/selinux/rego/src_test.rego [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    "object": {
+        "metadata": {
+            "name": "nginx"
+        },
+        "spec": {
+            "containers": input_containers_many,
+      }
+    }
+}
+
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/selinux/rego/src_test.rego:127]: As securityContext can be defined at both Pod and Container level, can you add some tests for this scenario?
https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-container

File: Policies/Kubernetes/selinux/rego/src_test.rego [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    results := violation with input as input
+    count(results) == 0
+}
+
+test_input_seLinux_options_two_allowed_in_list {
+    input := { "review": input_review_two, "parameters": input_parameters_in_list}
+    results := violation with input as input
+    count(results) == 0
+}
+
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/selinux/rego/src_test.rego:38]: Can you add tests with below combinations?
1. input_review_two and input_parameters_in_list_subset
1. input_review_two and input_parameters_empty

File: Policies/Kubernetes/selinux/constraint.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+metadata:
+  name: psp-selinux
+spec:
+  match:
+    excludedNamespaces: {{ .Values.excludedNamespaces }}
+    kinds:
+      - apiGroups: [""]
+        kinds: ["Pod"]
+  parameters:
+    allowedSELinuxOptions:
```
[the remark this answers is not in the document]
Comment (EmandM) [reply Policies/Kubernetes/selinux/constraint.yaml:12]: I would prefer to wait until we have better support for object types. I feel this is a good approach for now, and we can re-tool this policy if the requirements change in the future

[the remark this answers is not in the document]
Comment (RamyasreeChakka) [reply Policies/Kubernetes/selinux/constraint.yaml:12]: If this is the only policy which has object type, I would agree with you. But as we already have some other policies with object type and we are planning to improve the object type functionality, its better not to have mix up.

[the remark this answers is not in the document]
Comment (EmandM) [reply Policies/Kubernetes/selinux/constraint.yaml]: Cool, swapped over to second example

**Pull request merged**